src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java

```diff
@@ -0,0 +1,120 @@
+////////////////////////////////////////////////////////////////////////////////
+// Test case file for checkstyle.
+// Created: 2021
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocmethod;
+
+/* Config:
+ * scope = Scope.NOTHING.getName()
+ */
+public class InputMissingJavadocMethodPublicOnly2 // ignore - need javadoc
+{
+    private interface InnerInterface // ignore - when not relaxed about Javadoc
+    {
+        String CONST = "InnerInterface"; // ignore - w.n.r.a.j
+        void method(); // ignore - when not relaxed about Javadoc // ok
+
+        class InnerInnerClass // ignore - when not relaxed about Javadoc
+        {
+            private int mData; // ignore - when not relaxed about Javadoc
+
+            private InnerInnerClass() // ok
+            {
+                final Runnable r = new Runnable() {
+                    public void run() {}; // ok
+                };
+            }
+
+            void method2() // ignore - when not relaxed about Javadoc // ok
+            {
+                final Runnable r = new Runnable() {
+                    public void run() {};
+                };
+            }
+        }
+    }
+
+    private class InnerClass // ignore
+    {
+        private int mDiff; // ignore - when not relaxed about Javadoc
+
+        void method() // ignore - when not relaxed about Javadoc // ok
+        {
+        }
+    }
+
+    private int mSize; // ignore - when not relaxed about Javadoc
+    int mLen; // ignore - when not relaxed about Javadoc
+    protected int mDeer; // ignore
+    public int aFreddo; // ignore
+
+    // ignore - need Javadoc
+    private InputMissingJavadocMethodPublicOnly2(int aA) // ok
+    {
+    }
+
+    // ignore - need Javadoc when not relaxed
+    InputMissingJavadocMethodPublicOnly2(String aA) // ok
+    {
+    }
+
+    // ignore - always need javadoc
+    protected InputMissingJavadocMethodPublicOnly2(Object aA) // ok
+    {
+    }
+
+    // ignore - always need javadoc
+    public InputMissingJavadocMethodPublicOnly2(Class<Object> aA) // ok
+    {
+    }
+
+    // ignore - when not relaxed about Javadoc
+    private void method(int aA) // ok
+    {
+    }
+
+    // ignore - when not relaxed about Javadoc
+    void method(Long aA) // ok
+    {
+    }
+
+    // ignore - need javadoc
+    protected void method(Class<Object> aA) // ok
+    {
+    }
+
+    // ignore - need javadoc
+    public void method(StringBuffer aA) // ok
+    {
+    }
+
+
+    /**
+     A param tag should not be required here when relaxed about Javadoc.
+     Writing a little documentation should not be worse than not
+     writing any documentation at all.
+     */
```

Comment:
Just curious, why use the non-standard javadoc format here?

Reply:
Looks like I overlooked this, fixed now.

Reply:
Resolved.